### PR TITLE
core: add negative tests for htree

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Adds negative test for htree.

Tested-by: Jens Wiklander <jens.wiklander@linaro.org> (QEMU)
Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

These tests are intended to replace the entire regression_9xxx family in xtest. 
With this I believe that coverage is increased slightly and the tests themselves are more robust.

### CODE REVIEW

File: core/arch/arm/pta/core_fs_htree_tests.c [hunk 1]
```diff
@@ -449,17 +477,160 @@ TEE_Result core_fs_htree_tests(uint32_t nParamTypes,
...
+
+	res = test_get_offs_size(type, idx, 0, &offs, &size0);
+	CHECK_RES(res, return res);
+
+	aux2.data = malloc(aux->data_alloced);
+	if (!aux2.data)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	n = 0;
+	while (true) {
```
Comment (jforissier) [inline core/arch/arm/pta/core_fs_htree_tests.c]: Is this really testing that each single-byte corruption (at offset 0, 1, 2, 4 etc.) is detected? I don't think it does.

Comment (jenswi-linaro) [reply core/arch/arm/pta/core_fs_htree_tests.c]: Which part is it that you doubt?

Comment (jforissier) [reply core/arch/arm/pta/core_fs_htree_tests.c]: - If `do_range()` returns success for all values of `n` except the last one, the test will still pass.
- If `tee_fs_htree_open()` fails for some value of `n`, higher values won't be tested

Comment (jenswi-linaro) [reply core/arch/arm/pta/core_fs_htree_tests.c]: Thanks

**Comment by jenswi-linaro** [comment]
Updated with a fix of the test case.

**Comment by jforissier** [comment]
LGTM. I agree that white box testing in `xtest` is not very good, and is better replaced with lower-level tests inside optee_os when possible (and this is a good example). 

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
Squashed and tag applied

**Comment by jforissier** [comment]
The Travis failure seems to be compiler-dependent, and totally unrelated to this PR.
For the record:
- Travis uses [gcc-linaro-5.3-2016.02-x86_64_arm-linux-gnueabihf](http://releases.linaro.org/components/toolchain/binaries/5.3-2016.02/arm-linux-gnueabihf/gcc-linaro-5.3-2016.02-x86_64_arm-linux-gnueabihf.tar.xz) and fails. I can reproduce the issue with that compiler, even on the master branch.
- Shippable uses the Ubuntu 17.04 default cross-compiler (which is 6.3_something I believe), no issue
- I am using Ubuntu 16.04 with GCC 5.4.0-6ubuntu1~16.04.4 and that works too.

So we can safely ignore the failure as far as this PR is concerned.

**Pull request merged**